### DISCUSSION TITLE
struct or class should be using for string?

### DISCUSSION THREAD

**Discussion by arteny** [SteveDunn/Vogen#532] [Q&A] [answered]
In docs I found that all types are decalred as `partial struct` and only string are always declared as `partial class`.
But with my test i can use string as `partial struct` also.
So which is better to use and is there a difference between?

**Comment by SteveDunn** [comment, marked as answer]
Hi @arteny - just catching up on some Vogen action! Apologies for the delay!
Whether or not to use a class/struct value object is the same as deciding whether to use a class/struct for any other type.
Some other frameworks for wrapping primitives, wrapped them all in a class, which had performance implications, as, for every instance, there was now heap pressure on the GC.

Structs are a natural choice for value objects as they are small, have value-equality, and are immutable.

**Reply by fbestfriends** [reply]
I see, thanks, so struct for string is a good choice as I understood.
